---
output:
  md_document:
    variant: markdown_github
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

[![Build Status](https://travis-ci.org/3inar/geneset.svg?branch=master)](https://travis-ci.org/3inar/geneset)[![codecov](https://codecov.io/gh/3inar/geneset/branch/master/graph/badge.svg)](https://codecov.io/gh/3inar/geneset)


# geneset
Gene sets and functions for working with them.

### data sets
This package contains the following data sets:

* `smoking`: Blood transcriptome gene signatures that associate with cigarette 
  smoking from the Huan et al. 2016 meta-analysis http://dx.doi.org/10.1093/hmg/ddw288
  

# examples
### installation
```{r, eval=F}
devtools::install_github("3inar/geneset")
```

### load `.gmt` files
If you're working with gene sets from MSigDB, it's quite likely that you have files
in the Gene Matrix Transposed format; the `load_gmt()` function will read a `.gmt`
file into a `gset` object:
```{r}
library(geneset)
geneset <- load_gmt("tests/testthat/testgmt.gmt")  # dummy .gmt file for testing
geneset
```

### subset `gset` objects
You can subset `gset`s like you would a vector. There is also a `lenght` function
for them that returns the number of sets in the `gset`:
```{r}
geneset[2]
geneset[c(T, F, T)]
length(geneset)
```

### remove genes that aren't in your data
Inevitably some gene sets will contain symbols that for one reason or another
aren't present in the data set you're investigating. These can be removed by
`gsintersect()`:
```{r}
mygenes <- c("a", "b", "d", "e", "f")
geneset <- gsintersect(geneset, mygenes); geneset
```

### filter out gene sets that are too small or large
Perhaps a two-gene set is too small to be taken seriously for whatever reason,
`gsfilter()` will remove gene sets with cardinality outside of provided limits:
```{r}
geneset <- gsfilter(geneset, min=3); geneset
```
